tests/snmp/test_snmp_default_route.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+try:  # python3
+    from shlex import quote
+except ImportError:  # python2
+    from pipes import quote
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.device_type('vs')
+]
+
+
+@pytest.mark.bsl
+def test_snmp_default_route(duthosts, enum_dut_hostname, localhost, creds):
+    """compare the snmp facts between observed states and target state"""
+
+    duthost = duthosts[enum_dut_hostname]
+    hostip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
+    dut_result = duthost.shell('show ip route 0.0.0.0/0 | grep "\*"')
```

Comment:
You compare `show ip route` with snmp query. What is your expectation if either is empty or both are empty?

Reply:
show ip route might have no default route, default route just via eth0 or default route learned from bgp.
So based on the result, for each default route from show ip route, we will do a snmpget. 
For example:
vlab-01:~$ show ip route 0.0.0.0/0 | grep "\*"
  * 10.0.0.57, via PortChannel0001, weight 1
  * 10.0.0.59, via PortChannel0002, weight 1
  * 10.0.0.61, via PortChannel0003, weight 1
  * 10.0.0.63, via PortChannel0004, weight 1
   we will check if we are able to do snmpget on each of the above route:
   snmpget -v2c -c public 10.250.0.101  1.3.6.1.2.1.4.24.4.1.1.0.0.0.0.0.0.0.0.0.**10.0.0.57** 
iso.3.6.1.2.1.4.24.4.1.1.0.0.0.0.0.0.0.0.0.10.0.0.57 = IpAddress: 0.0.0.0
iso.3.6.1.2.1.4.24.4.1.1.0.0.0.0.0.0.0.0.0.10.0.0.57 = IpAddress: 0.0.0.0 
Here will check if the snmpget returns the string "ipaddress: 0.0.0.0" 
Also, 
snmpget -v2c -c public 10.250.0.101  1.3.6.1.2.1.4.24.4.1.16.0.0.0.0.0.0.0.0.0.**10.0.0.57**    
iso.3.6.1.2.1.4.24.4.1.16.0.0.0.0.0.0.0.0.0.10.0.0.57 = INTEGER: 1 -- status
Here will check if the above snmpget returns "integer: 1".
This way we will ensure that each of the "show ip route" route is retrieved via SNMP query also shows the correct result.

Reply:
Thanks for the example, I understand it is a case of "default route learned from bgp".

Should we explicitly cover more cases:
1. no default route
2.  default route just via eth0

Currently you did not check testing which case, so maybe a random one.

---
In reply to: [570694888](https://github.com/Azure/sonic-mgmt/pull/2934#discussion_r570694888) [](ancestors = 570694888)

Reply:
I have added test cases for both cases.